src/components/EnterBarcode/EnterBarcodePage.tsx

```diff
@@ -49,6 +49,10 @@ const EnterBarcodePage = ({
     }
   };
 
+  const handleFormSubmit = (e) => {
+    return barcode !== '' ? handleEnterBarcode() : e.preventDefault();
```

Comment:
just wanted to confirm, we have `e.preventDefault()` so the form doesn't submit if the barcode is empty, right?